File: Recursion/167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int start=0,end=n-1;
        int sum= numbers[start]+numbers[end];;
        while(start<end){  
            if(sum == target){
                return {start+1,end+1};
            } 
            else if(sum>target){
                sum-=numbers[end];
                end--;
                sum+=numbers[end];
            }
            else if(sum<target){
                sum-=numbers[start];
                start++;
                sum+=numbers[start];
            }
        }
        return {0,0};
    }
};

File: Binary_Tree/167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int n = numbers.length;

        int[] ans = {0,0};

        int start=0,end=n-1;

        while(start<end){

            int sum = numbers[start]+numbers[end];

            if(sum==target){
                ans[0] = start+1;
                ans[1] = end+1;
                return ans;
            }

            if(sum>target){
                end--;
            }
            else{
                start++;
            }


        }
        return ans;
        
    }
}